src/operations.jl

```diff
@@ -26,9 +37,21 @@ function Base.isequal(x::ArrayReference, y::ArrayReference)
     end
     true
 end
+"""
+    ArrayReferenceMeta
+
+A type similar to [`ArrayReference`](@ref) but holding additional information.
+
+# Fields
+
+$(TYPEDFIELDS)
+"""
 struct ArrayReferenceMeta
+    "The `ArrayReference`"
     ref::ArrayReference
+    "A vector of Bools indicating whether each index is a loop variable"
     loopedindex::Vector{Bool}
```

Comment:
If an index is computed (and not representable by an `offset`, e.g. if it's a function of two loop variables), an operation will be added, and the corresponding `loopedindex` set to `false`.